### ISSUE TITLE
Integrate services like Anilist or similar 

### ISSUE THREAD

**Issue by josueBarretogit** [josueBarretogit/manga-tui#22]

**Comment by justchokingaround** [comment]
would be great!

**Comment by crispy-caesus** [comment]
I'd also love this, only thing currently keeping me from using it

**Comment by josueBarretogit** [comment]
I'm going to be real with you all I dont know what anilist is XD like do you guys use it to keep a reading history? and if so how can I implement it? like when the user reads a chapter or downloads it then it is marked as read in their anilist account?

**Comment by josueBarretogit** [comment]
- [ ] Mark manga's chapter as read in anilist account when the user downloads or reads a chapter
- [ ] what else?

**Comment by crispy-caesus** [comment]
it would be nice, if you could select from anilist lists, like there's currently reading dropped and planning lists

**Comment by crispy-caesus** [comment]
it would be nice, if you could select from anilist lists, like there's currently reading dropped and planning lists

I'm not sure if downloaded should necessarily mark as read, maybe optionally if possible? I'm not sure if people maybe like to just download a few dozen chapters ahead and then read them slowly

**Comment by josueBarretogit** [comment]
Hi at the moment it's not ready for review, when it is I'll assign you and put instructions, thanks for wanting to help out

**Comment by josueBarretogit** [comment]
@crispy-caesus It is now ready for review I'll appreciate your input 🫶

**Comment by crispy-caesus** [comment]
Sorry for the late reply, I was a bit busy yesterday
```
Error: PlatformFailure(Dbus(D-Bus error: The name is not activatable (org.freedesktop.DBus.Error.ServiceUnknown)))
```
I am getting this error after inputting the token,I got the same error in a debug build, though it does seem this is not a problem debug would fix.
Trying anilist check after, also throws an error accordingly

I'm not sure if this is a problem with my system or what this exactly means, I'd love to help more but yeah I got no clue about Rust

**Comment by josueBarretogit** [comment]
Ok can you tell me what os you are using? in order to store de access token I use a secret service which may not be installed on your system, if you are on linux try running `sudo apt install libdbus-1-dev pkg-config`

**Comment by crispy-caesus** [comment]
ahhh alright, I'll start that first then, makes sense

**Comment by crispy-caesus** [comment]
ahh yes thank you, now it works perfectly!!

**Comment by crispy-caesus** [comment]
other things that could maybe be added would be having some kind of access to your lists from anilist, so that you can just access your reading or planning to read lists from the home page, just like the recently added maybe, or have it be a search parameter.

when adding the app, I don't personally know how exactly it is handled by others, but all other apps I have, are added to anilist as an app, where you are essentially provided a link and then authorize them, instead of creating your own api endpoint.

those are of course just small quality of life things that could and I think should just be added as seperate issues and PRs

**Comment by josueBarretogit** [comment]
> other things that could maybe be added would be having some kind of access to your lists from anilist, so that you can just access your reading or planning to read lists from the home page, just like the recently added maybe, or have it be a search parameter.

Sounds like a good idea, at the moment I want to integrate more services since I made it generic (dependency injection btw) so it should be easy

> when adding the app, I don't personally know how exactly it is handled by others, but all other apps I have, are added to anilist as an app, where you are essentially provided a link and then authorize them, instead of creating your own api endpoint.

Oh I didn't know you could do that 

> those are of course just small quality of life things that could and I think should just be added as seperate issues and PRs

Thanks for taking your time in testing out this new feature, could you please comment LGTM in the pr so that I can assign you (and approve the pr please)

**Comment by crispy-caesus** [comment]
> Thanks for taking your time in testing out this new feature, could you please comment LGTM in the pr so that I can assign you (and approve the pr please)

will do :+1: 


should I create separate issues for the other two things or how would you like to handle that?

**Comment by josueBarretogit** [comment]
> will do 👍

 I cant assign anybody i dont know why that is
 
> should I create separate issues for the other two things or how would you like to handle that?

yeah those should be separate issues ill make them, thanks 🙏

**Comment by crispy-caesus** [comment]
> I cant assign anybody i dont know why that is
it seems I would need some sort write access or pr/issue management perms to be able to be assigned

**Comment by crispy-caesus** [comment]
yo I'm just wondering, it only adds to anilist for whatever you read (download) in manga-tui. so only one way, not two way sync to also set read marker, at least chapter wise from anilist to manga-tui right now right?

**Comment by josueBarretogit** [comment]
> yo I'm just wondering, it only adds to anilist for whatever you read (download) in manga-tui. so only one way, not two way sync to also set read marker, at least chapter wise from anilist to manga-tui right now right?

Yes, it should work both ways right? yeah makes sense it worked both ways

**Comment by umop3plsdn** [comment]
im having the same Error: PlatformFailure(Dbus(D-Bus error: The name is not activatable (org.freedesktop.DBus.Error.ServiceUnknown))) on Arch but dbus is running systemctl status dbus
● dbus-broker.service - D-Bus System Message Bus
     Loaded: loaded (/usr/lib/systemd/system/dbus-broker.service;
disabled; preset: disabled)
     Active: active (running) since Fri 2025-02-28 07:34:17 EST; 3h 2min ago

**Comment by crispy-caesus** [comment]
that sounds like you don't have a secret service runnig, like kwallet, gnome wallet or keepassxc

**Comment by josueBarretogit** [comment]
> im having the same Error: PlatformFailure(Dbus(D-Bus error: The name is not activatable (org.freedesktop.DBus.Error.ServiceUnknown))) on Arch but dbus is running systemctl status dbus ● dbus-broker.service - D-Bus System Message Bus Loaded: loaded (/usr/lib/systemd/system/dbus-broker.service; disabled; preset: disabled) Active: active (running) since Fri 2025-02-28 07:34:17 EST; 3h 2min ago

make an issue for this

and this is closed so please make new issues instead

**Comment by josueBarretogit** [comment]
Actually im locking this rn

![Image](https://github.com/user-attachments/assets/8927fd6f-e9aa-4596-afa4-1f60a19fe986)
